server/server.go

```diff
@@ -615,24 +704,18 @@ func (s *Server) v1DataPost(w http.ResponseWriter, r *http.Request) {
 		if explainMode == types.ExplainFullV1 {
 			result.Explanation = types.NewTraceV1(*buf)
 		}
+		s.logDiagnostics(path.String(), goInput, nil, m, buf, settings)
```

Comment:
Nit: diagnostic readers can't differentiate between a `null` response and an *undefined* response.